addons/pos_loyalty/models/pos_config.py

```diff
@@ -73,12 +73,10 @@ def _check_before_creating_new_session(self):
 
     def use_coupon_code(self, code, creation_date, partner_id):
         self.ensure_one()
-        # Ordering by partner id to use the first assigned to the partner in case multiple coupons have the same code
-        #  it could happen with loyalty programs using a code
         # Points desc so that in coupon mode one could use a coupon multiple times
         coupon = self.env['loyalty.card'].search(
-            [('program_id', 'in', self._get_program_ids().ids), ('partner_id', 'in', (False, partner_id)), ('code', '=', code)],
-            order='partner_id, points desc', limit=1)
+            [('program_id', 'in', self._get_program_ids().ids), ('code', '=', code)],
+            order='points desc', limit=1)
```

Comment:
The relaxation should only be applied to gift cards. For potentially other coupons, partner_id should still be part of the criteria.

May I know the types of coupons that go thru this method? If it includes 'ewallets', then we can't apply this change.